whole-tale/girder_wholetale
# Add support for DV 4.16+ external tools

Starting with 4.16 Dataverse supports integration of external tools on a dataset level (see [docs](http://guides.dataverse.org/en/latest/installation/external-tools.html)). It comes with an introduction of additional query parameters that we now need to handle.

This PR introduces support for `{filePid}`, `{datasetId}` and `{datasetPid}`.

Additionally, I fixed a minor bug wrt PID protocol handling (we assumed it's "doi:" which is not True).

Related to https://github.com/whole-tale/wt-design-docs/pull/96 and https://github.com/whole-tale/wt-design-docs/pull/100

## How to test?

1. Deploy this branch locally, make sure `https://dev2.dataverse.org` is added to DATAVERSE_EXTRA_HOSTS (it's default in `setup_girder.py` if you're using `deploy-dev`)
1. Navigate to https://dev2.dataverse.org/dataset.xhtml?persistentId=doi:10.5072/FK2/VKI4P5
1. Click on Explore / Whole Tale (girder.local) (see image below)
   ![dv](https://user-images.githubusercontent.com/352673/65913245-2ad8e300-e395-11e9-9e1f-925e260709f5.png)
1. New page with WT should open. If you're not logged in, click on "Access WholeTale" and log in using any credentials
1. "Create New Tale" dialog should pop up with the following fields pre-populated:
   * Title: *Open Source at Harvard - File Hierarchy*
   * Input Data: *Data Source: https://dev2.dataverse.org/dataset.xhtml?persistentId=doi:10.5072/FK2/VKI4P5*
1. Choose you compute env an launch the Tale
1. Confirm that 'workspace' contains:
```
└──
    ├── code
    │   └── language-count.sh
    ├── data
    │   ├── 2019-02-25.tab
    │   └── 2019-02-25.tsv
    ├── results
    │   ├── language-count.tab
    │   └── language-count.tsv
    └── run.sh

4 directories, 6 files
```
